src/standalone/Vegetation/optimality_farquhar.jl

```diff
@@ -82,46 +82,40 @@ ClimaLand.auxiliary_domain_names(::OptimalityFarquharModel) =
     (:surface, :surface, :surface, :surface)
 
 """
-    update_photosynthesis!(
-        Rd,
-        An,
-        Vcmax25,
-        model::OptimalityFarquharModel,
-        T,
-        f_abs,
-        β,
-        medlyn_factor,
-        c_co2,
-        R,
-        energy_per_mole_photon_par,
-        par_d,
-    )
+    update_photosynthesis!(p, Y, t, model::OptimalityFarquharModel,canopy)
+
 Computes the net photosynthesis rate `An` for the Optimality Farquhar model, along with the
 dark respiration `Rd`, and the value of `Vcmax25`, and updates them in place.
-
- To do so, we require the canopy leaf temperature `T`, Medlyn factor, fraction of
-PAR radiation absorbed `f_abs`, incoming PAR radiation `par_d` in W/m^2,
- CO2 concentration in the atmosphere,
-moisture stress factor `β` (unitless), and the universal gas constant
-`R`.
-
-The typical `energy_per_mole_photon_par` is used to convert from an absorbed energy
-flux to a flux of moles of photons, as needed by photosynthetic rate computations.
 """
-function update_photosynthesis!(
-    Rd,
-    An,
-    Vcmax25,
-    model::OptimalityFarquharModel,
-    T,
-    f_abs,
-    β,
-    medlyn_factor,
-    c_co2,
-    R,
-    energy_per_mole_photon_par,
-    par_d,
-)
+function update_photosynthesis!(p, Y, t, model::OptimalityFarquharModel, canopy)
+    # unpack a bunch of stuff from p and params
+    Rd = p.canopy.photosynthesis.Rd
+    An = p.canopy.photosynthesis.An
+    Vcmax25 = p.canopy.photosynthesis.Vcmax25
+    T_canopy = canopy_temperature(canopy.energy, canopy, Y, p, t)
```

Comment:
If this function no longer takes `t`, we can remove `t` from the argument list of all of the `update` functions...

Reply:
It looks like the 2 implementations of `canopy_temperature` just return T from Y or p, so they don't really need t. Should we remove `t` from all the functions as part of this PR?

Reply:
yes! I can do that